src/WebJobs.Extensions.DurableTask/DurableOrchestrationContextBase.cs

```diff
@@ -454,6 +454,22 @@ public virtual Task CreateTimer(DateTime fireAt, CancellationToken cancelToken)
         /// <param name="input">The JSON-serializeable data to re-initialize the instance with.</param>
         public abstract void ContinueAsNew(object input);
 
+        /// <summary>
+        /// Restarts the orchestration by clearing its history.
+        /// </summary>
+        /// <remarks>
+        /// <para>Large orchestration histories can consume a lot of memory and cause delays in
+        /// instance load times. This method can be used to periodically truncate the stored
+        /// history of an orchestration instance.</para>
+        /// </remarks>
+        /// <param name="input">The JSON-serializeable data to re-initialize the instance with.</param>
+        /// <param name="preserveUnprocessEvents">
+        /// If set to <c>true</c>, re-adds any unprocessed external events into the new execution
+        /// history when the orchestration instance restarts. If <c>false</c>, any unprocessed
+        /// external events will be discarded when the orchestration instance restarts.
+        /// </param>
+        public virtual void ContinueAsNew(object input, bool preserveUnprocessEvents) => throw new NotImplementedException();
```

Comment:
What is our plan to document this new method? It seems using this new API is necessary to actually avoid the issue this PR sets out to address, but how will a customer know if they should use this new API or not?

Reply:
We'll need to rev the autogenerated API documentation after checking into `master`. I didn't want to make this the default behavior because it could lead to undesirable effects in some circumstances (e.g. a never-ending list of events that keep piling on). The main use case for this will be actors, which means our own code will need to call it. But we'll also suggest it to people who want to do actor-like things and preserve events.